peer_mentoring/groups/views.py

```diff
@@ -143,4 +144,38 @@ def reject_join_request(request, join_request_id):
     if group.moderator != request.user:
         raise PermissionDenied()
     join_request.delete()
-    messages.success(request, f"{join_request.user.username}'s join request")
+    messages.success(request, f"{join_request.user.username}'s join request was denied")
+
+
+@login_required
+def delete_post(request, group_id, post_id):
+    group = get_object_or_404(Group, id=group_id)
+    post = get_object_or_404(Post, id=post_id)
+    if group.moderator != request.user:
```

Comment:
nice, you can also abstract this using this decorator: 

<img width="781" alt="image" src="https://github.com/JAWD-001/peer_mentoring/assets/387927/99a6de3e-1f5b-4e5b-b705-8c1183e06a35">

Reply:
I can do this, would this be set up using a custom decorator like you did last week? I can just have them redirect back to the group_detail, or post. What do you think makes sense? Post? Also, what's the difference between a decorator and a generator? Was briefly looking around?

Reply:
A generator is a like a function but instead of returning values in one go, it "lazily" returns them when prompted with next() - see https://pybit.es/articles/generators/

A decorator is a piece of logic you want to apply to multiple functions or classes, see example above where @user_passes_test "enriches" the functions it decorates (it's applied to), this is away to abstract behavior in one place to not repeat yourself (keep it DRY)

Anyways, can you try the code example above returned by GPT and redirect to group index if the user does not pass the permission test? If stuck I can give you some more concrete code, but please try it ...